qiita_ware/dispatchable.py

```diff
@@ -5,48 +5,29 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
-from traceback import format_exception_only
-from sys import exc_info
-
-from .processing_pipeline import StudyPreprocessor, StudyProcessor
 from .analysis_pipeline import RunAnalysis
 from qiita_ware.commands import submit_EBI, submit_VAMPS
-from qiita_db.study import Study
+from qiita_ware.executor import execute
+from qiita_db.user import User
+from qiita_db.software import Parameters, DefaultParameters
 from qiita_db.analysis import Analysis
-from qiita_db.metadata_template.prep_template import PrepTemplate
+from qiita_db.artifact import Artifact
 
 
-def processor(preprocessed_data_id, param_id, param_constructor):
+def processor(user_id, preprocessed_data_id, param_id):
     """Dispatch the processor work"""
-    preprocessed_data = PreprocessedData(preprocessed_data_id)
-    params = param_constructor(param_id)
-
-    sp = StudyProcessor()
-    try:
-        process_out = sp(preprocessed_data, params)
-    except Exception as e:
-        error_msg = ''.join(format_exception_only(e, exc_info()))
```

Comment:
I think I just don't remember anymore, but I wanted to check with you that the exception is being caught and formatted properly like it was being done here?